lib/ansiblelint/utils.py

```diff
@@ -237,13 +237,13 @@ def _taskshandlers_children(basedir, k, v, parent_type):
             append_children(th['import_playbook'], basedir, k, parent_type, results)
         elif 'import_tasks' in th:
             append_children(th['import_tasks'], basedir, k, parent_type, results)
-        elif 'import_role' in th:
+        elif 'include_role' in th or 'import_role' in th:
             th = normalize_task_v2(th)
-            results.extend(_roles_children(basedir, k, [th['action'].get('name')], parent_type,
-                                           main=th['action'].get('tasks_from', 'main')))
-        elif 'include_role' in th:
-            th = normalize_task_v2(th)
-            results.extend(_roles_children(basedir, k, [th['action'].get('name')],
+            role_name = th['action'].get("name", None)
```

Comment:
Honestly, I'd prefer try/except KeyError, because checking for magic values is error-prone. Also, you'll get "Failed to find required key" even if it exists but is of a falsy type (`[]`, `{}`, `0`, `null`) which makes it confusing/misleading for users who try to debug it based on an error message produced.